### ISSUE TITLE
Macronix instead of Winbond chip?

### ISSUE THREAD

**Issue by outdoormagic** [tylernguyen/x1c6-hackintosh#116]
Hi everyone,

I read issue #68 and watched the videos. When I try to read the BIOS, I get this:

```
Calibrating delay loop... OK.
Found Macronix flash chip "MX25L12805D" (16384 kB, SPI) on ch341a_spi.
Found Macronix flash chip "MX25L12835F/MX25L12845E/MX25L12865E" (16384 kB, SPI) on ch341a_spi.
Multiple flash chip definitions match the detected chip(s): "MX25L12805D", "MX25L12835F/MX25L12845E/MX25L12865E"
Please specify which chip definition to use with the -c <chipname> option.

```

Is that normal? @tylernguyen I have the same config as you.

Could I be clipping the adapter onto the chip wrong? I do have the red wire on the 1 of the adapter, seated all the way back, and then the red wire on the clip aligned with the dot on the chip.

Thanks.

Paul

**Comment by tylernguyen** [comment]
@outdoormagic 

Are you clipping onto the right chip? Where is the chip you're clipping onto located?

**Comment by outdoormagic** [comment]
Yep. This is indeed the chip I'm clipping on to. As reference points, I am looking at the tape, the corner of the wifi antenna, the wire, and the silver/white square/silver arrangement on the other side of it.

The model is20KH-CT01WW.
÷÷÷÷÷÷÷

**Comment by tylernguyen** [comment]
> 
> 
> Yep. This is indeed the chip I'm clipping on to. As reference points, I am looking at the tape, the corner of the wifi antenna, the wire, and the silver/white square/silver arrangement on the other side of it.
> 
> The model is20KH-CT01WW.
> 


Hm.. I never flashed that chip before. So what you're doing is uncharted territory. 

> I do find that clipping the clamp on the chip is rather tricky. Perhaps my technique is wrong?
The clip is always going to be tricky on any chip, try placing it on one side then finishing the clip on the other.

Let me know how it goes flashing this chip. Remember to extract it twice and diff before modding.

**Comment by outdoormagic** [comment]
@tylernguyen 

Got a new programmer and clip.

I had to unscrew the wifi antenna next to the chip to secure the clip.

The chip is indeed a Macronix. Turns out they are a supplier for Winbond. The chip I have appears to be https://www.macronix.com/Lists/Datasheet/Attachments/7396/MX25L12873F,%203V,%20128Mb,%20v1.2.pdf, which is not directly supported by flashrom. I used the -c option to choose the "long" option that flashrom read (see original post):

`flashrom -p ch341a_spi -c MX25L12835F/MX25L12845E/MX25L12865E -r bios1.img`

Went through the mods, updated BIOS settings, and added framebuffer patch.

As a twist, an APFS sector map corruption set in (likely unrelated to the mod; just bad luck), so I ended up formatting the whole drive and setting up a clean Big Sur (no Windows 10 dual boot) system. I am now running the latest Big Sur with a modded system and Windows 10 Pro booting off another SSD in the WWAN slot. I also have a 512GB SD card in the reader slot (usually my photo backup).

Thanks again for your help. I hope the chip info helps others.
